lib/kafka/ssl_socket_with_timeout.rb

```diff
@@ -163,6 +163,10 @@ def closed?
       @tcp_socket.closed? || @ssl_socket.closed?
     end
 
+    def to_io
```

Comment:
seems good. Also we can probably swing you a testing cluster somehow, if you're interested.